modules/dxp/apps/analytics/analytics-reports-layout-impl/src/main/java/com/liferay/analytics/reports/layout/internal/info/item/LayoutAnalyticsReportsInfoItem.java

```diff
@@ -0,0 +1,209 @@
+import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;
+import com.liferay.portal.kernel.service.LayoutLocalService;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.ServiceContextThreadLocal;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.service.permission.LayoutPermissionUtil;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.PropsKeys;
+import com.liferay.portal.kernel.util.PropsUtil;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author David Arques
+ */
+@Component(service = AnalyticsReportsInfoItem.class)
+public class LayoutAnalyticsReportsInfoItem
+	implements AnalyticsReportsInfoItem<Layout> {
+
+	@Override
+	public String getAuthorName(Layout layout) {
+		return _getUserOptional(
+			layout
+		).map(
+			User::getFullName
+		).orElse(
+			StringPool.BLANK
+		);
+	}
+
+	@Override
+	public long getAuthorUserId(Layout layout) {
+		return _getUserOptional(
+			layout
+		).map(
+			User::getUserId
+		).orElse(
+			0L
+		);
+	}
+
+	@Override
+	public WebImage getAuthorWebImage(Layout layout, Locale locale) {
+		Optional<User> userOptional = _getUserOptional(layout);
+
+		return userOptional.map(
+			user -> {
+				try {
+					return new WebImage(
+						user.getPortraitURL(_getThemeDisplay()));
+				}
+				catch (PortalException portalException) {
+					_log.error(portalException, portalException);
+
+					return new WebImage(StringPool.BLANK);
+				}
+			}
+		).orElse(
+			new WebImage(StringPool.BLANK)
+		);
+	}
+
+	@Override
+	public List<Locale> getAvailableLocales(Layout layout) {
+		return Stream.of(
+			layout.getAvailableLanguageIds()
+		).map(
+			LocaleUtil::fromLanguageId
+		).collect(
+			Collectors.toList()
+		);
+	}
+
+	@Override
+	public Locale getDefaultLocale(Layout layout) {
+		return LocaleUtil.fromLanguageId(layout.getDefaultLanguageId());
+	}
+
+	@Override
+	public Date getPublishDate(Layout layout) {
+		return layout.getPublishDate();
+	}
+
+	@Override
+	public String getTitle(Layout layout, Locale locale) {
+		return layout.getTitle(locale);
+	}
+
+	@Override
+	public boolean isShow(Layout layout) {
+		if (!layout.isTypeContent()) {
+			return false;
+		}
+
+		if (_isEmbeddedPersonalApplicationLayout(layout)) {
+			return false;
+		}
+
+		try {
+			if (!_hasEditPermission(
+					layout, PermissionThreadLocal.getPermissionChecker())) {
+
+				return false;
+			}
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException, portalException);
+
+			return false;
+		}
+
+		return true;
+	}
+
+	private ThemeDisplay _getThemeDisplay() {
```

Comment:
Could we avoid extracting the `themeDisplay` in this way? Also take into account that it could be null. Maybe it'd make sense to have a InfoItemFieldValuesProvider for Layout.

Reply:
No problem, we don't have another option :(, this is the same way we do it in the InfoItem framework.

Reply:
I will check the null about it. Thank you very much :)